daemons/dss-dlq-reaper/app.py

```diff
@@ -0,0 +1,68 @@
+import json
+import logging
+import os
+import sys
+
+import boto3
+import domovoi
+
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), 'domovoilib'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss.logging import configure_lambda_logging
+from dss.util.aws import send_sns_msg
+
+logger = logging.getLogger(__name__)
+configure_lambda_logging()
+
+DSS_REAPER_RETRY_KEY = 'DSS-REAPER-RETRY-COUNT'
+# Max number of retries per message before we give up
+DSS_MAX_RETRY_COUNT = 10
+
+# Number of times messages are picked from the DLQ per execution (per minute)
+RECEIVE_BATCH_COUNT = 10
+
+app = domovoi.Domovoi(configure_logs=False)
+sqs = boto3.resource('sqs')
+
+
+@app.scheduled_function("rate(1 minute)")
+def reaper(event, context):
+    queue_name = "dss-dlq-" + os.environ["DSS_DEPLOYMENT_STAGE"]
+
+    # Get the queue
+    queue = sqs.get_queue_by_name(QueueName=queue_name)
+
+    message_count = 0
+
+    while context.get_remaining_time_in_millis() > 10000:
```

Comment:
You should account for the time it takes to long poll *and* to process each of the max. of ten messages. IOW, leave a little bit of head room.

Reply:
Agreed, maybe make this 20000.